### PR TITLE
[js/webgpu] Use builtin num_workgroups to fix shader key conflict

### PR DESCRIPTION
**Pull request by axinging**
This fixes conformance failure of tinyyolov2-8 and potential shader key conflict issues.

### CODE REVIEW

**Comment by axinging** [comment]
@qjia7 @xhcao @hujiajie @gyagp PTAL

File: js/web/lib/wasm/jsep/backend-webgpu.ts [hunk 1]
```diff
@@ -54,20 +54,22 @@ const getProgramInputTensorInfoDependencyKey =
...
-      getProgramInputTensorInfoDependencyKey(
-          inputTensors,
-          programInfo.shaderCache?.inputDependencies ??
-              new Array<ProgramInputTensorInfoDependency>(inputTensors.length).fill('dims'))}`;
-  return key;
-};
+const getProgramInfoUniqueKey =
+    (programInfo: ProgramInfo, inputTensors: readonly TensorView[], is1DimensionDispatch: boolean): string => {
+      // final key format:
+      // <PROGRAM_NAME>[<PROGRAM_CUSTOM_CACHE_HINT>]:<INPUTS_INFO_0>|<INPUTS_INFO_1>|...
```
Comment (qjia7) [inline js/web/lib/wasm/jsep/backend-webgpu.ts:60]: nit: `// <PROGRAM_NAME>[<PROGRAM_CUSTOM_CACHE_HINT>]:is1DimensionDispatch:<INPUTS_INFO_0>|<INPUTS_INFO_1>|...`

**Review by qjia7 — APPROVED** [review]
LGTM with one nit. Thanks.

File: js/web/lib/wasm/jsep/backend-webgpu.ts (hunk 1 above)
Comment (axinging) [reply js/web/lib/wasm/jsep/backend-webgpu.ts]: done

**Comment by qjia7** [comment]
@guschmue @fs-eire @satyajandhyala Please take a look, thanks.

**Review by fs-eire — APPROVED** [review]

**Pull request merged**